File: ExamSystem_BE/JWT/DTOs/Auth/RegisterRequest.cs
using System.ComponentModel.DataAnnotations;

namespace JWT.DTOs.Auth
{
    public class RegisterRequest
    {
        [Required]
        [MaxLength(100)]
        public string FullName { get; set; } = string.Empty;

        [Required]
        [EmailAddress]
        [MaxLength(255)]
        public string Email { get; set; } = string.Empty;

        [Required]
        [MaxLength(100)]
        public string Username { get; set; } = string.Empty;

        [Required]
        [MinLength(6)]
        [RegularExpression(
            @"^(?=.*[A-Za-z])(?=.*\d)(?=.*[^A-Za-z\d]).{6,}$",
            ErrorMessage = "Mật khẩu phải có ít nhất 6 ký tự, bao gồm chữ, số và ký tự đặc biệt.")]
        public string Password { get; set; } = string.Empty;

        /// <summary>
        /// Role đăng ký: "Student" (default) hoặc "Teacher".
        /// Admin không được phép tự đăng ký.
        /// </summary>
        [MaxLength(50)]
        public string Role { get; set; } = "Student";
    }
}

File: examsystem_fe/lib/features/auth/models/register_request.dart
/// Map từ RegisterRequest.cs trong BE:
/// FullName, Email, Username, Password (>=6 ký tự, có chữ + số + ký tự đặc biệt), Role
class RegisterRequest {
  final String fullName;
  final String email;
  final String username;
  final String password;

  /// Role đăng ký: 'Student' (default) hoặc 'Teacher'.
  final String role;

  final String? avatarUrl;

  const RegisterRequest({
    required this.fullName,
    required this.email,
    required this.username,
    required this.password,
    this.role = 'Student',
    this.avatarUrl,
  });

  Map<String, dynamic> toJson() => {
    'fullName': fullName,
    'email':    email,
    'username': username,
    'password': password,
    'role':     role,
    if (avatarUrl != null) 'avatarUrl': avatarUrl,
  };
}
